far/viewer.cpp

```diff
@@ -4089,3 +4159,80 @@ void Viewer::vgetc_cache::compact()
 	m_End = std::copy(m_Iterator, m_End, m_Buffer);
 	m_Iterator = m_Buffer;
 }
+
+#ifdef ENABLE_TESTS
+
+#include "testing.hpp"
+
+TEST_CASE("hex.mode.bytes.area.width")
+{
+	struct test_data
+	{
+		int Bytes;
+		int Width;
+	};
+
+	static const test_data TestDataPoints[] =
+	{
+		{  8,  24 }, // > 2D 2D 2D 2D 2D 2D 2D 2D<
+		{  9,  29 }, // > 2D 2D 2D 2D 2D 2D 2D 2D │ 2D<
+		{ 10,  32 }, // > 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D<
+		{ 13,  41 }, // > 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D<
+		{ 15,  47 }, // > 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D<
+		{ 16,  50 }, // > 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D<
+		{ 17,  55 }, // > 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D │ 2D<
+		{ 20,  64 }, // > 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D<
+		{ 24,  76 }, // > 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D<
+		{ 25,  81 }, // > 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D │ 2D<
+		{ 30,  96 }, // > 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D<
+		{ 32, 102 }, // > 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D<
+		{ 33, 107 }, // > 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D │ 2D<
+	};
+
+	for (const auto& TestDataPoint : TestDataPoints)
+	{
+		REQUIRE(TestDataPoint.Width == hex_mode_bytes_area_wdth(TestDataPoint.Bytes));
+	}
+}
+
+TEST_CASE("hex.mode.bytes.from.line.width")
+{
+	for (const auto CharSize : { 1, 2 })
+	{
+		for (const auto& Bytes : std::views::iota(8, 260))
+		{
+			for (const auto& LineWidth : std::views::iota(hex_mode_line_width_from_bytes(Bytes, CharSize), hex_mode_line_width_from_bytes(Bytes + 1, CharSize)))
+			{
+				REQUIRE(Bytes == hex_mode_bytes_from_line_width(LineWidth, CharSize));
+			}
+		}
+	}
+}
+
+int HexModeByteFromColumn16BytesFixed(const int Column, const int CharSize)
```

Comment:
Please mark as static to avoid "[missing declaration](https://github.com/FarGroup/FarManager/actions/runs/8609968009/job/23594740225?pr=825#step:7:249)" warnings

Reply:
Of course! Fixed.